src/target/imxrt.c

```diff
@@ -172,10 +172,10 @@ bool imxrt_probe(target_s *const target)
 	target->target_options |= CORTEXM_TOPT_INHIBIT_NRST;
 	target->driver = "i.MXRT10xx";
 
-#if defined(ENABLE_DEBUG) && PC_HOSTED == 1
+#if defined(ENABLE_DEBUG)
 	const uint8_t boot_mode = (target_mem_read32(target, IMXRT_SRC_BOOT_MODE2) >> 24U) & 3U;
-#endif
 	DEBUG_TARGET("i.MXRT boot mode is %x\n", boot_mode);
```

Comment:
This macro should resolve to a noop on not-BMDA so should cause boot_mode to be considered unused (which is a -Werror'd warning). That suggests the `#if` check should still check for `PC_HOSTED == 1` as all DEBUG_TARGET() output only actually exists in BMDA.

Reply:
On Farpatch we send `DEBUG_TARGET()` to the ESP-IDF logging stream, which is helpful.

Is it not desirable to enable debug prints on a build that is not PC_HOSTED?

Reply:
Ah, that's actually very helpful if you have the Flash space, yes. The platforms that are in-repo do not because we do not have sufficient Flash available on the majority of the targets, so DEBUG_ERROR(), DEBUG_WARN() and DEBUG_INFO() are the only 3 that get included in an `ENABLE_DEBUG=1` firmware binary usually as even that uses vast amounts of space and requires disabling targets to get to fit.

This is also how the in-repo machinery in src/include/general.h works:

https://github.com/blackmagic-debug/blackmagic/blob/799a4088e6c98fcbd977d9c3f2036bef4ba1e9b6/src/include/general.h#L74-L76

https://github.com/blackmagic-debug/blackmagic/blob/799a4088e6c98fcbd977d9c3f2036bef4ba1e9b6/src/include/general.h#L86-L90

We'll need to include some more complicated `#if` logic to allow Farpatch to continue including DEBUG_TARGET() in firmware and we weren't aware it was doing that to avoid breaking it.

Reply:
This is how it's done in Farpatch:

https://github.com/farpatch/farpatch/blob/835ba05bcf92e42cfa0ccbb28bf6a0fa6be5f232/components/blackmagic/general.h#L71-L86

(I have to reimplement the ESP macros here but without LF since the BMP logging system always appends `\n` as part of their logging macros, but aside from that it works well.)

How would you like this patch to change?

Reply:
Because they need to expand as block statements, we suggest https://github.com/farpatch/farpatch/blob/835ba05bcf92e42cfa0ccbb28bf6a0fa6be5f232/components/blackmagic/general.h#L89-L92 gets modified to use PRINT_NOOP(), as an aside.

Regarding this patch, we're going to suggest as the guard:

```c
#if defined(ENABLE_DEBUG) && (PC_HOSTED == 1 || defined(ESP_LOGW_BMP))
```

We think that'll prevent problems in building the firmware on the in-repo platforms while preserving the behaviour Farpatch needs.

Reply:
`ESP_LOGW_BMP` is somewhat of an implementation detail, and I hope it will go away when a new logging mechanism is introduced in BMP.

I've changed it to `ESP_LOGD`, which is the macro for debugging, and exists in all ESP builds.